qark/scanner/scanner.py

```diff
@@ -36,115 +42,31 @@ def run(self):
-                plugin.run(files=self.files, apk_constants={"minimum_sdk": get_min_sdk(self.decompiler.manifest_path),
-                                                            "target_sdk": get_target_sdk(
-                                                                self.decompiler.manifest_path)})
-            except Exception:
-                log.exception("Error running plugin %s... continuing with next plugin", plugin_name)
-                continue
-
-            self.issues.extend(plugin.issues)
-
-    def _run_broadcast_checks(self):
-        """
-        Runs all plugins under `qark.plugins.broadcast` and updates `self.issues` with their findings.
-        """
-        plugin_source = get_plugin_source(category="broadcast")
-        for plugin_name in get_plugins(category="broadcast"):
-            try:
-                plugin = plugin_source.load_plugin(plugin_name).plugin
-            except Exception:
-                log.exception("Error loading plugin %s... continuing with next plugin", plugin_name)
-                continue
-
-            try:
-                plugin.run(files=self.files, apk_constants={"minimum_sdk": get_min_sdk(self.decompiler.manifest_path),
-                                                            "target_sdk": get_target_sdk(
-                                                                self.decompiler.manifest_path)})
-            except Exception:
-                log.exception("Error running plugin %s... continuing with next plugin", plugin_name)
-                continue
-
-            self.issues.extend(plugin.issues)
-
-    def _run_file_checks(self):
-        """
-        Runs all plugins under `qark.plugins.file` and updates `self.issues` with their findings.
-        """
-        plugin_source = get_plugin_source(category="file")
-        for plugin_name in get_plugins(category="file"):
-          try:
-            plugin = plugin_source.load_plugin(plugin_name).plugin
-          except Exception:
-            log.exception("Error loading plugin %s... continuing with next plugin", plugin_name)
-            continue
-
-            try:
-                plugin.run(files=self.files, apk_constants={"minimum_sdk": get_min_sdk(self.decompiler.manifest_path),
-                                                            "target_sdk": get_target_sdk(
-                                                                self.decompiler.manifest_path)})
-            except Exception:
-                log.exception("Error running plugin %s... continuing with next plugin", plugin_name)
-                continue
+        plugin_source = get_plugin_source(category=category)
+        try:
+            min_sdk = get_min_sdk(self.decompiler.manifest_path, files=self.files)
+            target_sdk = get_target_sdk(self.decompiler.manifest_path, files=self.files)
+        except AttributeError:
+            # manifest path is not set, assume min_sdk and target_sdk
+            min_sdk = target_sdk = 15
```

Comment:
Where did you get 15 for the default from? Last docs I saw had `1` as the minsdk if it wasn't set.

Reply:
Ah, forgot to make a note about this above, thanks for catching. This is when the manifest file is not present in the scan (like a single java file being scanned). I didn't know what to default it to since 1 seemed too low, so I picked a middle range.

Some checks only create issues based off of these min_sdk and target_sdk so deciding what to do when we do not have one (no manifest.xml) is up for discussion.